### PR TITLE
CCM-6250: shared modules

### PR DESCRIPTION
**Pull request by sidnhs**
## Description
Making use of shared modules in new shared-modules github repo(https://github.com/NHSDigital/nhs-notify-shared-modules/)

Deploy: https://github.com/NHSDigital/nhs-notify-internal/actions/runs/10527676559

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -61,3 +61,6 @@ tests/test-team/test-results/
 tests/test-team/playwright-report/
 tests/test-team/blob-report/
 tests/test-team/playwright/.cache/
+
+#plugin-cache
```
Comment (aidenvaines-bjss) [inline .gitignore:65]:
```suggestion

```

File: infrastructure/terraform/components/branch/module_amplify_branch.tf [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-  amplify_app_id                            = local.iam.amplify["id"]
-  branch                                    = var.branch_name
-  domain_name                               = local.root_domain_name
-  subdomain                                 = var.environment
-  base_path                                 = "/auth~${local.normalised_branch_name}"
+  description    = "Amplify branch for ${local.normalised_branch_name}"
+  amplify_app_id = local.iam.amplify["id"]
+  branch         = var.branch_name
+  display_name   = local.normalised_branch_name
+  environment_variables = {
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/branch/module_amplify_branch.tf:14]: youll need to run the terraform fmt again

Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/branch/module_amplify_branch.tf:14]: makes way more sense to have this in env vars, nice

File: infrastructure/terraform/components/iam/module_amplify_branch.tf [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-  cognito_user_pool_client_id               = aws_cognito_user_pool_client.main.id
-  cognito_user_pool_identity_provider_names = aws_cognito_user_pool_client.main.supported_identity_providers
-  amplify_app_id                            = aws_amplify_app.main.id
-  branch                                    = "main"
-  domain_name                               = local.root_domain_name
-  subdomain                                 = var.environment
-  base_path                                 = "/auth"
-  enable_auto_deploy                        = true
-  stage                                     = "PRODUCTION"
+  source            = "git::https://github.com/NHSDigital/nhs-notify-shared-modules.git//infrastructure/modules/amp_branch?ref=v1.0.0"
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/iam/module_amplify_branch.tf:2]: put an empty line after the meta attributes like source,
```suggestion
source = "git::https://github.com/NHSDigital/nhs-notify-shared-modules.git//infrastructure/modules/amp_branch?ref=v1.0.0"

```

File: infrastructure/terraform/components/iam/module_amplify_branch.tf [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
+  name              = "main"
+  display_name      = "main"
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/iam/module_amplify_branch.tf:4]: Where's display_name come from?

**Review by aidenvaines-bjss — CHANGES_REQUESTED** [review]

File: infrastructure/terraform/components/branch/module_amplify_branch.tf (hunk 2 above)
Comment (sidnhs) [reply infrastructure/terraform/components/branch/module_amplify_branch.tf:14]: weirdly, running terraform fmt doesn't seem to think it needs to change  this 🤔

File: .gitignore (hunk 1 above)
Comment (sidnhs) [reply .gitignore:65]: I have done that now

Comment (sidnhs) [reply .gitignore:65]: oh actually, doesn't let me push: 
`remote: error: See https://gh.io/lfs for more information.
remote: error: File plugin-cache/registry.terraform.io/hashicorp/aws/5.64.0/darwin_amd64/terraform-provider-aws_v5.64.0_x5 is 551.40 MB; this exceeds GitHub's file size limit of 100.00 MB`

Comment (sidnhs) [reply .gitignore:65]: sorry my bad, done

File: infrastructure/terraform/components/iam/module_amplify_branch.tf (hunk 4 above)
Comment (sidnhs) [reply infrastructure/terraform/components/iam/module_amplify_branch.tf:4]: removed it, doesn't seem to give much benefit

**Pull request merged**